webapp/channels/src/components/dialog_router/interactive_dialog_adapter.tsx

```diff
@@ -0,0 +1,844 @@
+                if (option.value === null || option.value === undefined || String(option.value).trim() === '') {
+                    errors.push({
+                        field: `${fieldPrefix}.options[${optIndex}].value`,
+                        message: `${optionType} value is required`,
+                        code: 'REQUIRED',
+                    });
+                }
+            }
+        }
+
+        // Validation for text fields with proper server limits
+        if (element.type === 'text' || element.type === 'textarea') {
+            // Validate min/max length relationship
+            if (element.min_length !== undefined && element.max_length !== undefined) {
+                if (element.min_length > element.max_length) {
+                    errors.push({
+                        field: `${fieldPrefix}.min_length`,
+                        message: 'min_length cannot be greater than max_length',
+                        code: 'INVALID_FORMAT',
+                    });
+                }
+            }
+
+            // Validate against server limits
+            const maxLengthLimit = element.type === 'textarea' ? 3000 : 150;
+            if (element.max_length !== undefined && element.max_length > maxLengthLimit) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > ${maxLengthLimit} (server limit for ${element.type})`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
+
+        // Validation for select fields
+        if (element.type === 'select') {
+            // Warn if both options and data_source are provided
+            if (element.options && element.data_source) {
+                errors.push({
+                    field: `${fieldPrefix}.options`,
+                    message: 'Select element cannot have both options and data_source',
+                    code: 'INVALID_FORMAT',
+                });
+            }
+
+            // Validate max_length for select fields (server limit is 3000)
+            if (element.max_length !== undefined && element.max_length > 3000) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > 3000 (server limit for select)`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
+
+        // Validation for bool fields
+        if (element.type === 'bool') {
+            // Validate max_length for bool fields (server limit is 150)
+            if (element.max_length !== undefined && element.max_length > 150) {
+                errors.push({
+                    field: `${fieldPrefix}.max_length`,
+                    message: `max_length too large: ${element.max_length} > 150 (server limit for bool)`,
+                    code: 'TOO_LONG',
+                });
+            }
+        }
```

Comment:
What is the meaning of max_length in a boolean or select field? Shouldn't they be ignored in those cases?